mlir/include/mlir/Dialect/MIOpen/Pipeline.h

```diff
@@ -0,0 +1,36 @@
+//===- Pipeline.h - MIOpen pipeline creation ---------------===//
+//
+// Part of the MLIR Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file declares the miopen::addPipeline routine to consolidate the
+// MIOpen pipeline creation.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef MLIR_DIALECT_MIOPEN_PIPELINE_H
+#define MLIR_DIALECT_MIOPEN_PIPELINE_H
+
+#include <mlir/Pass/PassManager.h>
+
+#include <string>
+
+namespace mlir {
+namespace miopen {
+
+// Compilation pipeline from MIXR/TOSA to MIOpen
+void addHighLevelPipeline(PassManager &pm);
+
+// Compilation pipeline from MIOpen to LLVM
+void addPipeline(PassManager &pm, const std::string &perfConfig="", bool applicability=false, bool highLevel=false);
+
+// Compilation pipeline from LLVM to Binary
+void addBackendPipeline(PassManager &pm, const std::string &triple, const std::string &chip, const std::string &features, int32_t optLevel=3);
```

Comment:
I like those much better than the previous version to use another config to control the flow of pipelines. Thanks

Reply:
As long as a you call in the right order... 🥇